lib/rouge/lexers/cmhg.rb

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class CMHG < RegexLexer
+      title "CMHG"
+      desc "RISC OS C module header generator source file"
+      tag 'cmhg'
+      filenames '*.cmhg'
+
+      def self.preproc_keyword
+        @preproc_keyword ||= %w(
+          define elif else endif error if ifdef ifndef include line pragma undef warning
+        )
+      end
+
+      state :root do
+        rule %r/;[^\n]*/, Comment
+        rule %r/^[ \t]*#[ \t]*(?:(?:#{CMHG.preproc_keyword.join('|')})[ \t].*)?/, Comment::Preproc
+        rule %r/[-a-z]+:/, Keyword::Declaration
```

Comment:
Is this supposed to be case sensitive?

Reply:
Yes - everything about this file format is case-sensitive, apart from the `x` in the hexadecimal number prefix. I could have listed the currently defined keywords, but the following colon unambiguously marks them out as a keyword, in a manner that adapts if new keywords are introduced in future.